File: design-patterns/go/mediator/general/colleague.go
// colleague.go

package main

type IColleague interface {
	ReceiveMessage(message string)
	SendMessage(colleague IColleague, message string)
}

type Colleague struct {
	mediator *Mediator
}

func NewColleague(mediator *Mediator) (colleague *Colleague) {
	colleague = &Colleague{}
	colleague.mediator = mediator
	return
}

File: design-patterns/php/mediator/general/Colleague.php
<?php
// Colleague.php

namespace BigBoxCode\DesignPattern\Mediator\General;

abstract class Colleague {
    public function __construct(protected Mediator $mediator) {
    }

    abstract function sendMessage(Colleague $colleague, string $message): void;
    abstract function receiveMessage(string $message): void;
}